models/articles.model.js

```diff
@@ -32,4 +32,21 @@ function fetchArticles() {
     });
 }
 
-module.exports = { fetchArticleById, fetchArticles };
+function fetchCommentsByArticleId(article_id) {
+  return db
+    .query(
+      `SELECT comment_id, votes, created_at, author, body, article_id 
+      FROM comments 
```

Comment:
For this task in the SQL statement, we are selecting data from the comments table where the comments.article_id matches the one specified in the url. So right now on line 45-46 we are rejecting if rows.length === 0. 

In other words, if there are no comments returned from the select query, we are rejecting with a 404 and a message saying article does not exist. However this doesn't make sense here, as rows refers to the list of comments returned from our query, not if the article exists or not. 

The difficulty is that we could get an empty array for if the article exists but has no comments AND if the article doesn't exist, hence the need for some kind of check to discern between the two scenarios. 

When the article exists but has no comments we would expect a 200, when the article requested does not exist then we'd expect a 404. In short, this reject should happen elsewhere for example in a utility function that checks if the article exists elsewhere. Check out the notes here for some pointers! You may remember David using Promise.all to accomplish run both asynchronous functions together as opposed to the usage of await here

https://notes.northcoders.com/courses/js-back-end/advanced-error-handling

Reply:
I'll take a look at this thank you!